lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -167,6 +167,67 @@ static Value createSoftmaxBackwardCommonKernel(PatternRewriter &rewriter,
   return sub;
 }
 
+namespace {
+/// We decompose aten.amax into a set of aten.max.dim op(s) depending on the
+/// number of dimensions across which the max needs to be computed.
+/// Eg:
+///   INPUT:
+///      final_output = aten.amax(initial_input, (0, 2), keepdim=False)
+///
+///   OUTPUT:
+///      input_1 = aten.max.dim(initial_input, 0, keepdim)  #1
+///      final_output = aten.max.dim(input_1, 1, keepdim)   #2
+///
+/// NOTE: In the above example aten.max.dim #2 is sent 1 as the `dim` instead of
+///       2 because after #1 the 0-th dimension from the `initial_input` is
+///       erased.
+///       This is because `keepdim` was set as False. Had it been True, we would
+///       have sent 2 as the reducing dimension even in the second invocation of
+///       #2.
+///       Therefore for every dimension provided in aten.amax, we create a new
+///       aten.max.dim AND for every such dimension (except the first one) we'd
+///       make an adjustment (refer the code comment) before calling
+///       aten.max.dim.
+class DecomposeAtenAmaxOp : public OpRewritePattern<AtenAmaxOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenAmaxOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    SmallVector<Value> dims;
+    if (!getListConstructElements(op.dim(), dims))
+      return rewriter.notifyMatchFailure(
+          op, "unimplemented: dims not list of Scalar");
+
+    bool keepDim;
+    if (!matchPattern(op.keepdim(), m_TorchConstantBool(&keepDim)))
+      return rewriter.notifyMatchFailure(
+          op, "Expected a constant boolean value for keepDim");
+
+    Value input = op.self();
+    // For every dimension included in `dim` of the op, we create a call to
+    // aten.max.dim.
+    for (unsigned i = 0; i < dims.size(); i++) {
+      int64_t cstDim = -1;
+      if (!matchPattern(dims[i], m_TorchConstantInt(&cstDim)))
+        return rewriter.notifyMatchFailure(
+            op, "unimplemented: dim must be constant");
+      // We adjust the reducing dimension in case `keepDim` is unset.
+      if (!keepDim)
+        cstDim -= i;
```

Comment:
This does not handle all inputs correctly. Namely, it does not take into account repeated dim values and dim values out of order. I would recommend first sorting the `dims` argument and removing any duplicates, then iterating over the `dims` argument in reverse order. In reverse order no adjustment needs to take place. 

 You should add e2e tests for those two cases as well

Reply:
This was really helpful @ramiro050 ! It irked me when the "adjustment" I was making was exactly same as the "inverse" of the loop IV! Should've seen that it's effectively doing what reverse iterating would've. Haha. Thank you so much for this! :D

Regarding the two test cases:
I did address it in the code and added two initially but the case with duplicate/repeated dim values bails out at the PyTorch level itself.
So, I've taken off the unnecessary check of the duplicate values from the code and now the update revision only adds one more test case to the bunch which tests for random ordering of the supplied dim values.